scripts/install_plugins.sh

```diff
@@ -5,32 +5,84 @@ CURRENT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 source "$CURRENT_DIR/shared_functions.sh"
 
 clone() {
-	local plugin=$1
-	cd $SHARED_TPM_PATH &&
+	local plugin; plugin="$1"
+	cd "$SHARED_TPM_PATH" &&
 		GIT_TERMINAL_PROMPT=0 git clone --recursive $plugin
 }
 
+install_plugin_git() {
+	local plugin; plugin="$1"
+	local branch; branch=":${1##*:}"
+	plugin="${1%$branch}"
+	clone "$plugin" ||
+		 clone "git:$plugin"
+	if [ "${branch}" != ":${1}" ]; then #if exists branch/revision
+		(cd "$SHARED_TPM_PATH"/$(shared_plugin_name "$plugin") &&
+		git checkout -q "${branch#:}" 2>&1 >/dev/null)
```

Comment:
Sorry, I don't see where it could be added, the cd sub-command would fail if no such directory exists